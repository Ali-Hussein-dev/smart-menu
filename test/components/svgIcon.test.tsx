import { render } from '@testing-library/react'
import React from 'react'
import {
  SvgIcon,
  GetMenuType,
  GetDishes,
  GetDrinks,
} from '../../src/components/leaf-node'

describe('SVG Icon', () => {
  it('dishes menu matches snapshot', () => {
    const { container } = render(
      <SvgIcon>
        <GetMenuType iconName="dishes" />
      </SvgIcon>
    )
    expect(container).toMatchSnapshot()
  })
  it('drinks menu matches snapshot', () => {
    const { container } = render(
      <SvgIcon>
        <GetMenuType iconName={'drinks'} />
      </SvgIcon>
    )
    expect(container).toMatchSnapshot()
  })
  it('burger icon matches snapshot', () => {
    const { container } = render(
      <SvgIcon>
        <GetDishes iconName={'burger'} />
      </SvgIcon>
    )
    expect(container).toMatchSnapshot()
  })
  it('softdrinks icon matches snapshot', () => {
    const { container } = render(
      <SvgIcon>
        <GetDrinks iconName={'softdrinks'} />
      </SvgIcon>
    )
    expect(container).toMatchSnapshot()
  })
})
