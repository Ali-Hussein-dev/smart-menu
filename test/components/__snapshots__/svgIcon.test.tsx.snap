// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVG Icon burger icon matches snapshot 1`] = `
<div>
  <div
    class="fill-current w-10 h-10 "
  >
    <svg
      viewBox="0 -3 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M294.672 63.957c32.84 5.086 63.3 16.64 88.09 33.41 23.972 16.219 41.597 36.54 50.972 58.77a9.997 9.997 0 009.22 6.12 9.973 9.973 0 003.882-.788c5.09-2.149 7.473-8.012 5.328-13.102-10.883-25.808-31.008-49.172-58.195-67.566-27.219-18.414-60.496-31.074-96.239-36.61-5.449-.843-10.566 2.895-11.41 8.356-.847 5.453 2.89 10.562 8.352 11.41zm0 0M100.352 138.973c1.582 3.797 5.609 6.355 9.718 6.16 4.059-.196 7.785-3.067 9.02-6.938 1.25-3.918-.149-8.492-3.43-10.988-3.394-2.578-8.383-2.672-11.867-.21-3.707 2.616-5.156 7.792-3.441 11.976zm0 0M148.531 101.734c1.598 3.84 5.696 6.426 9.856 6.16 4.011-.261 7.66-3.12 8.89-6.945 1.254-3.89-.152-8.422-3.379-10.93-3.41-2.644-8.484-2.753-11.984-.218-3.64 2.636-5.117 7.773-3.383 11.933zm0 0M200.29 138.973c1.581 3.797 5.612 6.355 9.722 6.16 4.058-.196 7.793-3.067 9.027-6.938 1.25-3.922-.156-8.496-3.441-10.988-3.395-2.578-8.38-2.672-11.868-.21-3.703 2.616-5.156 7.792-3.44 11.976zm0 0"
      />
      <path
        d="M512 316.938c0-22.02-14.148-40.786-33.824-47.72v-25.023a42.946 42.946 0 009.074-4.629c11.387-7.652 17.918-19.73 17.918-33.148 0-8.39-2.594-16.332-7.414-23.012a32.269 32.269 0 001.988-18.511c-8.562-45.797-37.969-87.72-82.8-118.051C372.288 16.637 315.131 0 256 0S139.71 16.637 95.059 46.844c-44.832 30.328-74.239 72.254-82.801 118.05a32.247 32.247 0 004.176 22.774c-3.114 5.71-4.735 12.07-4.735 18.75 0 13.418 6.531 25.504 17.922 33.152 1.36.914 2.77 1.73 4.203 2.477v27.172C14.148 276.152 0 294.918 0 316.938v2.03c0 14.587 6.215 27.743 16.125 36.985a39.225 39.225 0 00-4.426 18.121c0 10.012 3.64 19.274 10.215 26.512v36.086c0 38.55 31.363 69.918 69.918 69.918h328.336c38.555 0 69.918-31.367 69.918-69.918v-31.55c9.621-7.669 15.082-18.798 15.082-31.052 0-7.683-2.164-14.957-6.207-21.246C507.059 343.855 512 331.98 512 318.97zm-20 0v2.03c0 10.325-5.148 19.466-13.008 25.009a30.65 30.65 0 01-17.578 5.578H306.348l63.77-63.203h91.292c16.867 0 30.59 13.722 30.59 30.586zm-340.434-30.586l1.586 1.59 10.258 10.253-21.473 21.477-33.32-33.32zm94.418 66.132l-61.351-61.347c0-.004-.004-.008-.008-.012l-4.773-4.773h161.812l-66.832 66.093c-.004.004-.008.008-.012.016l-.027.023-14.406 14.407zM31.914 168.566c7.574-40.5 33.98-77.847 74.352-105.16C147.645 35.414 200.82 20 256 20s108.355 15.414 149.734 43.406c40.371 27.313 66.774 64.66 74.352 105.16 1.402 7.512-4.598 14.688-12.223 14.688H44.137c-7.637 0-13.625-7.176-12.223-14.688zm.523 32.512a32.564 32.564 0 0011.7 2.176h423.726c5.52 0 11.07-1.43 15.883-4.137a19.387 19.387 0 011.422 7.297c0 6.645-3.305 12.676-9.07 16.547-5.895 3.96-13.403 4.91-20.145 2.777-3.332-1.054-6.476-2.414-9.914-3.148-3.719-.797-7.523-1.149-11.324-1.149-17.863 0-27.45 6.766-35.149 12.2-6.757 4.773-12.093 8.543-23.609 8.543s-16.852-3.77-23.61-8.54c-7.699-5.44-17.28-12.203-35.148-12.203-17.863 0-27.445 6.766-35.144 12.204-6.758 4.77-12.094 8.539-23.61 8.539-11.515 0-16.855-3.77-23.613-8.543-7.7-5.438-17.281-12.2-35.144-12.2s-27.45 6.766-35.149 12.2c-6.758 4.773-12.098 8.543-23.61 8.543-11.519 0-16.855-3.77-23.616-8.543-7.7-5.438-17.282-12.2-35.149-12.2-7.066 0-13.437 1.2-19.48 3.664-7.137 2.915-15.489 2.176-21.91-2.14-5.766-3.871-9.07-9.903-9.07-16.547-.005-1.84.245-3.629.734-5.34zm23.375 45.738c5.247-.254 10.168-1.578 15.051-3.437 4.871-1.86 10.184-2.227 15.344-1.766 8.828.785 13.656 4.196 19.57 8.367 7.7 5.438 17.282 12.204 35.149 12.204 17.863 0 27.445-6.766 35.144-12.204 6.758-4.77 12.098-8.539 23.614-8.539s16.851 3.77 23.609 8.54c7.7 5.437 17.285 12.203 35.148 12.203 17.864 0 27.446-6.766 35.145-12.204 6.758-4.77 12.098-8.539 23.61-8.539 11.515 0 16.855 3.77 23.613 8.54 7.699 5.437 17.28 12.203 35.144 12.203s27.445-6.766 35.145-12.204c6.761-4.77 12.097-8.539 23.613-8.539 4.086 0 8.098.68 11.906 2.176 3.715 1.461 7.582 2.512 11.555 2.977v19.761H53.824v-19.484a48.252 48.252 0 001.989-.055zM20 316.937c0-16.863 13.723-30.585 30.586-30.585h29.75l54.531 54.535c1.875 1.875 4.418 2.925 7.07 2.925s5.196-1.05 7.07-2.925l28.548-28.547 37.215 37.215H50.586a30.604 30.604 0 01-15.125-4.02C26.234 340.265 20 330.335 20 318.97zm13.352 49.59a50.736 50.736 0 0017.234 3.028H234.77l18.546 18.547c3.836 3.84 10.262 3.847 14.11.03l18.746-18.577H461.41c5.64 0 11.274-.957 16.602-2.801 1.117-.387 2.258-.777 3.308-1.317.34-.175.832-.566 1.235-.539.554.04.699.747.898 1.192 1.082 2.387 1.66 5 1.707 7.62.168 9.435-6.625 17.33-15.496 19.9-5.082 1.476-10.586 1.148-15.484-.852-6.032-2.465-12.403-3.664-19.47-3.664-17.862 0-27.444 6.765-35.144 12.203-6.761 4.773-12.097 8.543-23.613 8.543-11.512 0-16.851-3.77-23.61-8.543-7.698-5.438-17.28-12.203-35.144-12.203s-27.445 6.765-35.148 12.203c-6.758 4.773-12.094 8.543-23.61 8.543s-16.851-3.77-23.613-8.543c-7.7-5.438-17.281-12.203-35.144-12.203-17.864 0-27.446 6.765-35.149 12.203-6.758 4.773-12.094 8.543-23.61 8.543-11.519 0-16.855-3.774-23.613-8.543-7.703-5.438-17.285-12.203-35.152-12.203-7.066 0-13.437 1.199-19.476 3.668-6.528 2.664-14.137 2.312-20.297-1.157-5.45-3.066-9.508-8.324-10.461-14.57a19.733 19.733 0 01-.055-5.594c.07-.539.988-5.093 1.48-4.914zm436.734 70.145c0 27.523-22.395 49.918-49.918 49.918H91.832c-27.523 0-49.918-22.395-49.918-49.918v-23.719c8.211 2.328 17.074 2.164 25.52-.644a45.21 45.21 0 002.809-1.031c3.593-1.47 7.491-2.18 11.917-2.18 11.52 0 16.856 3.765 23.613 8.539 7.704 5.437 17.286 12.203 35.153 12.203 17.863 0 27.445-6.766 35.144-12.203 6.758-4.774 12.098-8.54 23.61-8.54 11.515 0 16.855 3.766 23.613 8.54 7.7 5.437 17.281 12.203 35.148 12.203 17.864 0 27.446-6.766 35.145-12.203 6.758-4.774 12.098-8.54 23.61-8.54 11.515 0 16.855 3.766 23.609 8.54 7.699 5.437 17.285 12.203 35.148 12.203s27.445-6.766 35.145-12.203c6.757-4.774 12.097-8.54 23.613-8.54 2.98 0 5.969.333 8.848 1.133 2.738.762 5.324 1.965 8.062 2.739 5.945 1.676 12.363 2.031 18.461 1.039v22.664zm0 0"
      />
      <path
        d="M225.91 101.734c1.582 3.797 5.613 6.375 9.727 6.164 4.058-.207 7.781-3.066 9.02-6.933 1.257-3.922-.153-8.488-3.43-10.992-3.387-2.59-8.383-2.688-11.864-.215-3.691 2.617-5.199 7.793-3.453 11.976zm0 0M271.02 145.813c1.582 3.8 5.609 6.355 9.718 6.16 4.063-.196 7.778-3.07 9.024-6.934 1.261-3.918-.153-8.496-3.434-10.992-3.394-2.582-8.383-2.668-11.867-.211-3.703 2.617-5.156 7.793-3.441 11.976zm0 0M245.488 56.023c3.559 2.372 8.563 2.137 11.871-.574 3.235-2.648 4.528-7.34 3.036-11.254-1.454-3.8-5.305-6.507-9.387-6.488-4.04.023-7.88 2.676-9.309 6.457-1.597 4.234.035 9.336 3.79 11.86zm0 0"
      />
    </svg>
  </div>
</div>
`;

exports[`SVG Icon dishes menu matches snapshot 1`] = `
<div>
  <div
    class="fill-current w-10 h-10 "
  >
    <svg
      clip-rule="evenodd"
      fill-rule="evenodd"
      stroke-linejoin="round"
      stroke-miterlimit="2"
      viewBox="0 0 2134 2134"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1800 233.333h-166.667V100c-.009-18.28-15.053-33.316-33.333-33.316-1.449 0-2.897.094-4.333.283L328.997 233.633c-1.526.368-3.02.859-4.467 1.467a33.3603 33.3603 0 00-5.766 1.933 32.5736 32.5736 0 00-5.467 3.667 37.4098 37.4098 0 00-4.467 3.733 35.3015 35.3015 0 00-3.733 5.567 30.1502 30.1502 0 00-2.733 5 33.7766 33.7766 0 00-1.467 7.167 33.5867 33.5867 0 00-.9 4.5v1766.67c0 18.286 15.047 33.334 33.333 33.334H1800c18.286 0 33.333-15.048 33.333-33.334V266.667c0-18.286-15.047-33.334-33.333-33.334zm-233.333-95.3v95.3H842.2l724.467-95.3zm200 1861.97h-1400v-1700h1400v1700z"
        fill-rule="nonzero"
      />
      <path
        d="M1397.63 981.4l164.967-165-47.133-47.133-164.967 165-47.167-47.134 164.967-165-47.1-47.133-188.567 188.567c-55.774 55.794-64.668 143.92-21.166 209.733l-147.534 147.367-507.033-506.9c-13.147-12.433-33.986-12.433-47.133 0-49.587 49.822-54.76 129.482-12.034 185.3l275.867 358.566c6.303 8.229 16.097 13.064 26.462 13.064 8.822 0 17.292-3.501 23.538-9.73l93.067-93.167 100 100-279 279.3 47.133 47.133L1063.93 1315.1l279.167 279.133 47.133-47.133-279.133-279.167 147.5-147.5c65.821 43.493 153.948 34.585 209.733-21.2l188.567-188.566-47.133-47.134-165 164.967-47.134-47.1zm-594.3 205.5l-252.7-328.467c-14.874-19.445-19.22-45.034-11.6-68.3l330.5 330.534-66.2 66.233zm476.667-134.8c-29.919-29.889-37.788-75.67-19.567-113.833l133.334 133.333c-38.277 17.689-83.749 9.909-113.967-19.5h.2z"
        fill-rule="nonzero"
      />
    </svg>
  </div>
</div>
`;

exports[`SVG Icon drinks menu matches snapshot 1`] = `
<div>
  <div
    class="fill-current w-10 h-10 "
  >
    <svg
      clip-rule="evenodd"
      fill-rule="evenodd"
      stroke-linejoin="round"
      stroke-miterlimit="2"
      viewBox="0 0 2134 2134"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1800 233.333h-166.667V100c-.009-18.28-15.053-33.316-33.333-33.316-1.449 0-2.897.094-4.333.283L328.997 233.633c-1.526.368-3.02.859-4.467 1.467a33.3603 33.3603 0 00-5.766 1.933 32.5736 32.5736 0 00-5.467 3.667 37.4098 37.4098 0 00-4.467 3.733 35.3015 35.3015 0 00-3.733 5.567 30.1502 30.1502 0 00-2.733 5 33.7766 33.7766 0 00-1.467 7.167 33.5867 33.5867 0 00-.9 4.5v1766.67c0 18.286 15.047 33.334 33.333 33.334H1800c18.286 0 33.333-15.048 33.333-33.334V266.667c0-18.286-15.047-33.334-33.333-33.334zm-233.333-95.3v95.3H842.2l724.467-95.3zm200 1861.97h-1400v-1700h1400v1700z"
        fill-rule="nonzero"
      />
      <path
        d="M799.813 1356.42c0 185.866-1.825 221.333-1.959 223.654-1.146 17.183 4.85 34.087 16.584 46.713 11.825 12.72 28.408 19.95 45.779 19.95h412.908c17.371 0 33.954-7.23 45.779-19.95 11.729-12.625 17.729-29.53 16.579-46.713-.133-2.337-1.958-37.833-1.958-223.65 0-68.854 16.517-139.175 35.638-220.596 32.616-138.879 69.583-296.283 37.979-503.583-4.654-30.529-30.909-53.079-61.788-53.079H787.979c-30.879 0-57.133 22.554-61.787 53.079-31.604 207.3 5.362 364.704 37.979 503.583 19.125 81.417 35.642 151.738 35.642 220.592zm447.666-249.175c-20.033 85.283-38.95 165.841-38.95 249.175 0 78.204-.416 163.873-.012 198.685l-287.396-.106c.405-34.813 3.696-120.375 3.696-198.579 0-83.334-18.917-163.892-38.95-249.175-11.675-49.717-23.492-123.054-32.325-176.071l424.779 2.225c-8.833 53.016-19.167 124.129-30.842 173.846zM843.196 692.916l446.95.118c4.179 44.038 4.908 96.611 3.171 136.123H840.025c-1.733-39.508-1.008-92.2 3.171-136.241z"
        fill-rule="nonzero"
      />
    </svg>
  </div>
</div>
`;

exports[`SVG Icon softdrinks icon matches snapshot 1`] = `
<div>
  <div
    class="fill-current w-10 h-10 "
  >
    <svg
      viewBox="20 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M135.069 332.43c-1.859-1.86-4.439-2.93-7.069-2.93s-5.21 1.069-7.07 2.93c-1.86 1.861-2.93 4.44-2.93 7.07s1.069 5.21 2.93 7.069c1.86 1.861 4.44 2.931 7.07 2.931s5.21-1.07 7.069-2.931c1.86-1.859 2.931-4.439 2.931-7.069s-1.07-5.21-2.931-7.07z"
      />
      <path
        d="M393.041 109.726l-17.169-36.321c3.861-1.385 6.628-5.067 6.628-9.405 0-5.523-4.478-10-10-10h-76.366l4.419-26.135c1.175-6.946-.76-14.017-5.308-19.397C290.698 3.086 284.049 0 277.004 0h-42.008c-7.045 0-13.694 3.086-18.241 8.467-4.548 5.381-6.482 12.451-5.308 19.397L215.866 54H139.5c-5.522 0-10 4.477-10 10 0 4.338 2.767 8.02 6.629 9.405l-17.17 36.321A9.9982 9.9982 0 00118 114v181.5c0 5.522 4.478 10 10 10s10-4.478 10-10V124h167.667c5.522 0 10-4.477 10-10s-4.478-10-10-10H143.788l14.182-30h196.06L374 116.245V442H138v-64c0-5.522-4.478-10-10-10s-10 4.478-10 10v74c0 1.478.327 2.938.959 4.273l17.53 37.083c-2.98 1.732-4.989 4.95-4.989 8.644 0 5.522 4.478 10 10 10h229c5.522 0 10-4.478 10-10 0-3.694-2.009-6.912-4.988-8.644l17.529-37.083c.632-1.336.959-2.796.959-4.273V114c0-1.478-.327-2.937-.959-4.274zM280.832 24.53L275.85 54h-39.7l-4.982-29.47c-.262-1.546.428-2.639.862-3.153.434-.514 1.398-1.377 2.966-1.377h42.008c1.568 0 2.531.862 2.966 1.376.435.514 1.124 1.607.862 3.154zM354.03 492H157.97l-14.182-30h224.424l-14.182 30z"
      />
      <path
        d="M352.71 106.93c-1.851-1.86-4.431-2.93-7.07-2.93-2.63 0-5.2 1.07-7.07 2.93-1.859 1.86-2.93 4.44-2.93 7.07s1.07 5.21 2.93 7.07c1.87 1.86 4.44 2.93 7.07 2.93 2.639 0 5.219-1.07 7.07-2.93 1.87-1.86 2.939-4.44 2.939-7.07s-1.069-5.21-2.939-7.07zM258.667 27h-5.334c-5.522 0-10 4.477-10 10s4.478 10 10 10h5.334c5.522 0 10-4.477 10-10s-4.478-10-10-10zM256 199c-46.317 0-84 37.682-84 84 0 46.317 37.683 84 84 84s84-37.683 84-84c0-46.318-37.683-84-84-84zm0 148c-35.29 0-64-28.71-64-64s28.71-64 64-64 64 28.71 64 64-28.71 64-64 64z"
      />
    </svg>
  </div>
</div>
`;
